language/move-compiler/src/typing/expand.rs

```diff
@@ -25,8 +25,18 @@ pub fn function_body_(context: &mut Context, b_: &mut T::FunctionBody_) {
 pub fn function_signature(context: &mut Context, sig: &mut FunctionSignature) {
     for (_, st) in &mut sig.parameters {
         type_(context, st);
+        core::check_non_fun(context, st)
```

Comment:
These checks should really be in `typing/translate` and used at each declaration

Reply:
I check it here since all inferred types should be expanded. Currently, for signatures there is no inference, but for locals.